blockstore/splitstore/splitstore_compact.go

```diff
@@ -1079,27 +1132,43 @@ func (s *SplitStore) waitForMissingRefs(markSet MarkSet) {
 		}
 
 		towalk := missing
-		walked := cid.NewSet()
+		visitor := tmpVisitor()
 		missing = make(map[cid.Cid]struct{})
 
 		for c := range towalk {
-			err := s.walkObjectIncomplete(c, walked,
+			err := s.walkObjectIncomplete(c, visitor,
 				func(c cid.Cid) error {
 					if isUnitaryObject(c) {
 						return errStopWalk
 					}
 
-					mark, err := markSet.Has(c)
-					if err != nil {
-						return xerrors.Errorf("error checking markset for %s: %w", c, err)
-					}
-
-					if mark {
-						return errStopWalk
+					visitor, ok := markSet.(MarkSetVisitor)
```

Comment:
Can we somehow avoid this cast?

Reply:
Yes we can -- I can widen the `MarkSetVisitor` type to be `ObjectVisitor` + `MarkSet` and then we don't need to cast or bother with the else branch code.

Reply:
done.